### PR TITLE
GEODE-7414: Added Client Hello extension

### PR DESCRIPTION
**Pull request by mivanac**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [*] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [*] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [*] Is your initial contribution a single, squashed commit?

- [*] Does `gradlew build` run cleanly?

- [*] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java [hunk 1]
```diff
@@ -1076,6 +1080,20 @@ private void configureClientSSLSocket(Socket socket, int timeout) throws IOExcep
 
       SSLParameters modifiedParams =
           checkAndEnableHostnameValidation(sslSocket.getSSLParameters());
+
+      String sysId = "";
+      InternalDistributedSystem system = InternalDistributedSystem.getAnyInstance();
+      if (system != null) {
+        sysId = system.getConfig().getClientHelloExtension();
```
Comment (jinmeiliao) [inline geode-core/src/main/java/org/apache/geode/internal/net/SocketCreator.java]: 1. If this property is ssl related, it should start with "ssl-"
1. I don't see the connection between the 'client-hello-extension' and 'sysId', probably a better name for this property?
1. adding a config property probably would need a discussion thread on the dev list to gather feedback first.

**Comment by mivanac** [comment]
This is only solution with updated config parameter. Start working on Jens proposal.

**Comment by mivanac** [comment]
Idea is that ssl-server-name-extension will contain generic string value which will be passed as Server Name extension in Client Hello message.  Could you clarify your idea. What will be content of ssl-parameter-extension parameter?

**Comment by jdeppe-pivotal** [comment]
If the user wishes to manipulate `SSLParameter`s (say in order to add their own SNI value) they would need to implement an interface something like:
```
public interface SSLParameterExtension {
  SSLParameter modify(SSLParameter, SSLConfig);
}
```
On starting Geode, they would provide the following parameter:
```
ssl-parameter-extension=com.example.ssl.MySSLParams.class
```
When the `SSLConfig` class is created it would also create (and hold a reference to) an instance of this class. At the point where SSLParameters need to be adjusted, this class' `modify` method could be called in order to perform whatever changes the user wants.

Depending on your particular use case you may need to also have the interface provide an `init()` method with whatever parameters are necessary for `modify` to do what you want - perhaps `Cache` or `DistributionConfig` - something fairly general so that additional use cases would likely also be satisfied in the future.

**Pull request closed without merging**